### PR TITLE
[WIP] Add support to merge VCF files using samtools

### PR DESCRIPTION
**Pull request by xbrianh**
This currently requires installation of a slightly modified version of htslib and bcftools:
```
git clone -b xbrianh-readers-idx https://github.com/xbrianh/htslib.git
git clone -b xbrianh-omit-index https://github.com/xbrianh/bcftools.git
(cd bcftools && make)
(cd bcftools && sudo make install)
```

Either PRs with the above functionality will be made against [htslib](https://github.com/samtools/htslib) and [bcftools](https://github.com/samtools/bcftools), or those libraries will be wrapped into terra-notebook-utils somehow. In the meantime, the VCF merge workflows will be constructed with this branch.

### CODE REVIEW

File: scripts/merge_vcfs.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+parser = argparse.ArgumentParser()
+parser.add_argument("bucket", type=str, help="")
+parser.add_argument("output_key", type=str, help="")
+parser.add_argument("input_keys", help="GS bucket locations of '.vcf.gz' files.")
+args = parser.parse_args()
+
+keys = args.input_keys.split(",")
+vcf.combine(args.bucket, keys, args.bucket, args.output_key)
+
+# merge_vcfs.py "fc-9169fcd1-92ce-4d60-9d2d-d19fd326ff10" "combined.vcf.gz" "consent1/HVH_phs000993_TOPMed_WGS_freeze.8.chr7.hg38.vcf.gz,phg001280.v1.TOPMed_WGS_Amish_v4.genotype-calls-vcf.WGS_markerset_grc38.c2.HMB-IRB-MDS/Amish_phs000956_TOPMed_WGS_freeze.8.chr7.hg38.vcf.gz"
```
Comment (DailyDreaming) [inline scripts/merge_vcfs.py:19]: Maybe throw this as an example in the file docstring?

File: scripts/merge_vcfs.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Output to `output_key` in the same bucket.
+"""
+import argparse
+
+from terra_notebook_utils import vcf
+
+parser = argparse.ArgumentParser()
+parser.add_argument("bucket", type=str, help="")
+parser.add_argument("output_key", type=str, help="")
+parser.add_argument("input_keys", help="GS bucket locations of '.vcf.gz' files.")
```
Comment (DailyDreaming) [inline scripts/merge_vcfs.py:13]: Maybe include that it's comma-delimited in the `help`?

File: tests/test_terra_notebook_utils.py [hunk 3]
```diff
@@ -155,8 +155,35 @@ def test_prepare_merge_workflow_input(self):
         self.assertEqual(expected_input_keys, sorted(input_keys))
         self.assertEqual(expected_output_keys, sorted(output_keys))
 
+    def _test_combine_with_bucket_objs(self):
```
Comment (DailyDreaming) [inline tests/test_terra_notebook_utils.py:158]: Are these tests run?

Comment (xbrianh) [reply tests/test_terra_notebook_utils.py:160]: No, they must be run manually.

I've added TODOs to create fixtures for these tests.

File: scripts/merge_vcfs.py (hunk 1 above)
Comment (xbrianh) [reply scripts/merge_vcfs.py]: done

File: tests/test_terra_notebook_utils.py (hunk 3 above)
Comment (xbrianh) [reply tests/test_terra_notebook_utils.py:160]: done

File: scripts/merge_vcfs.py (hunk 2 above)
Comment (xbrianh) [reply scripts/merge_vcfs.py]: Done

**Comment by xbrianh** [comment]
This functionality has been moved to the [xsamtools](https://github.com/xbrianh/xsamtools) package.

**Pull request closed without merging**